src/PrestaShopBundle/Resources/config/services/bundle/mail.yml

```diff
@@ -0,0 +1,30 @@
+services:
+  _defaults:
+    public: true
+
+  prestashop.service.mail.mail_template_folder_catalog:
+    class: 'PrestaShopBundle\Service\Mail\MailTemplateFolderCatalog'
+    arguments:
+      - '%mail_themes_dir%'
+
+  prestashop.service.mail.mail_template_parameters_builder:
+    class: 'PrestaShopBundle\Service\Mail\MailTemplateParametersBuilder'
+    arguments:
+      -
+        emailPublicWebRoot: "@=service('prestashop.adapter.legacy.context').getRootUrl()"
+
+  prestashop.service.mail.mail_template_renderer:
+    class: 'PrestaShopBundle\Service\Mail\MailTemplateRenderer'
+    arguments:
+      - '@templating.engine.twig'
+      - '@prestashop.service.mail.mail_template_parameters_builder'
+
+  prestashop.service.mail.mail_template_generator:
+    class: 'PrestaShopBundle\Service\Mail\MailTemplateGenerator'
+    arguments:
+      - '@prestashop.service.mail.mail_template_folder_catalog'
+      - '@prestashop.service.mail.mail_template_renderer'
+    calls:
```

Comment:
looks like you always have a logger, then it should be injected in the constructor: don't you think?

Reply:
the logger is not mandatory, for example in tests you don't need it
I prefer this method because it keeps the constructor lighter